public/app/core/utils/richHistory.ts

```diff
@@ -238,14 +250,21 @@ export function isParsable(string: string) {
   return true;
 }
 
-export function createDataQuery(query: RichHistoryQuery, queryString: string, index: number) {
+export function createDataQuery(
```

Comment:
Same as with createUrl, if getQueryFromDisplayText passed, use it to create url. Otherwise just parse queryString.